### PR TITLE
WIP: Watchlists

### PR DESCRIPTION
**Pull request by matthewdias**
#276, #309

I'm basing this off of Notifications. Started with the UI, because that's what I know. Also, I'm working on a mac, so it seems like some of the file modes have changed. Please let me know if this is an issue. I'll squash commits at significant milestones.

I'm new to .NET, as well as DDD so please give me a heads up if there's anything I should be doing differently.

### CODE REVIEW

**Comment by markus101** [comment]
> Please let me know if this is an issue. I'll squash commits at significant milestones.

It looks like a number of files have changed, including sqlite dylibs, none of those changes should be included, if there is a valid reason for the changes they should be in a separate PR.

I take it by the UI you're planning to implement them in a similar fashion to notifications/download clients/indexers? Given trakt gives a lot more than a watch list it may not make sense to scope these to be called watched lists (I don't see a case where we would setup syncing watchlists and other elements from trakt as separate integrations).

File: src/UI/Settings/Watchlists/Add/WatchlistAddCollectionView.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+var ThingyAddCollectionView = require('../../ThingyAddCollectionView');
+var AddItemView = require('./WatchlistAddItemView');
+
+module.exports = ThingyAddCollectionView.extend({
+	itemView          : AddItemView,
```
Comment (markus101) [inline src/UI/Settings/Watchlists/Add/WatchlistAddCollectionView.js]: Use spaces instead of tabs (there are a few cases of this).

**Comment by markus101** [comment]
Another thing to keep in mind is the UI is the first thing that will change for v3, so I wouldn't spend a lot of time on it (beyond what is already done) to avoid having to redo it, though the v3 UI is far from ready and a major WIP.

**Comment by matthewdias** [comment]
>It looks like a number of files have changed, including sqlite dylibs, none of those changes should be included, if there is a valid reason for the changes they should be in a separate PR.

Yeah, I didn't change anything here, so don't know why they changed. I'll remove those changes, when I rebase.

>I take it by the UI you're planning to implement them in a similar fashion to notifications/download clients/indexers? Given trakt gives a lot more than a watch list it may not make sense to scope these to be called watched lists (I don't see a case where we would setup syncing watchlists and other elements from trakt as separate integrations).

I was planning on only implementing watchlists, but if there is a desire for deeper trakt integration, (I guess like adding obtained episodes to a `Collected` list?) then I could rename the section to `Lists` or something similarly generic. In #309 there is talk about getting more data such as user ratings, etc. I think this would mean a more drastic change in Sonarr, so it should probably be discussed properly beforehand.

>Another thing to keep in mind is the UI is the first thing that will change for v3, so I wouldn't spend a lot of time on it (beyond what is already done) to avoid having to redo it, though the v3 UI is far from ready and a major WIP.

Didn't know that was happening, but yeah, I'm stealing it straight from notifications, so it's not a big deal.

**Comment by markus101** [comment]
> In #309 there is talk about getting more data such as user ratings, etc. I think this would mean a more drastic change in Sonarr, so it should probably be discussed properly beforehand.

Yeah, it'd definitely be a bigger thing to tackle, but just the watch list aspect of it would be advantageous. I think the whole concept would be part of a bigger Connections discussion, since its another 3rd party integration (like the notifications are). The more I think about it the less I'm worried about what its called now, we can change that in the UI.

**Comment by iiznrml** [comment]
Has any progress been made on this?

**Comment by matthewdias** [comment]
I've put it on hold until v3 is done. If anyone else wants to take a stab at it feel free.

**Comment by markus101** [comment]
Closing pending v3 changes.

**Pull request closed without merging**